tensorflow_datasets/core/download/downloader_test.py

```diff
@@ -146,13 +146,17 @@ def test_no_headers(self):
     self.assertEqual(res, 'baz.zip')
 
   def test_headers(self):
-    cdisp = ('attachment;filename="hello.zip";'
-             'filename*=UTF-8\'\'hello.zip')
-    resp = _FakeResponse('http://foo.bar/baz.zip', b'content', headers={
-        'content-disposition': cdisp,
-    })
-    res = downloader._get_filename(resp)
-    self.assertEqual(res, 'hello.zip')
+    cdisps = [['attachment;filename*=UTF-8\'\'hello.zip','hello.zip'],
+             ['Attachment; filename=example.html','example.html'],
+             ['INLINE; FILENAME= "an example.html"','an example.html'],
+             ["attachment;filename*= UTF-8''%e2%82%ac%20rates.zip",'20rates.zip'],
```

Comment:
The filename is not properly decoded right ? The resulting filename should be `€ rates.zip`. This could be fixed in a future PR if you prefer

Reply:
Do we need to decode it? The original bug was related to the extention not being detected causing the extraction to fail.

Reply:
Which is why this could be fixed in a separate issue, but this is still an issue nonetheless.
I think it would be best to raise an error in this case, or not infer the filename, rather than silently recording a wrong filename.

Reply:
Oh got it. So a separate case for `filename*=` to decode it.

Reply:
I've imported the code and made some changes to return None when only the encoded filename is provided.
We might want to fix the filename*= case later if we encounter a case where we need it.

Anyway, this will be merged soon. Thank you for your contribution

Reply:
I got the encoded filename part done, so should i commit to this PR or should i do it in a new PR (after this is merged)?

Reply:
Please open a new PR after this one has been merged, as I've already imported merged this one. It should be merged on github soon.